Tensile/Source/client/main.cpp

```diff
@@ -355,6 +355,56 @@ namespace Tensile
             return args;
         }
 
+        size_t getMaxWorkspace(std::shared_ptr<MasterSolutionLibrary<ContractionProblem>>& library,
```

Comment:
Does this code create extra output?

I think we could get away with growing the workspace as needed, as we iterate.

Reply:
GPU buffer allocation is in DataInitialization.
So I do iteration in getMaxWorkspace before running iteration.
Do we have plan to do growing workspace in running iteration??

Reply:
If `DataInitialization::prepareGPUInputs()` also takes a solution, it could look at the current required workspace and grow it if necessary.

Reply:
https://github.com/ROCmSoftwarePlatform/Tensile/blob/develop/Tensile/Source/client/include/DataInitializationTyped.hpp#L181
Here shows it only allocate GPU buffer at first call with max buffer size.
Looks like, it didn't re-allocate GPU buffer for each problem which is good behavior for training.

If we really want to put max workspace size calculation in DataInitialization, we can pass solution set in MasterSolutionLibrary as input to DataInitialization and calculate max workspace on constructor of DataInitialization.
Do we need to do that??